### PR TITLE
feat(math): add conjugacy classes for permutation groups

### LINKED ISSUE #1664: [Feature]: Add bounded exact group invariants (order, element order, subgroup lattice)
## Summary

Jacobian has `finite_abelian_group.exact_factorization.compute` but no operations for basic finite group invariants: group order from a presentation, element order, subgroup lattice, or conjugacy class computation. These are fundamental building blocks for the group-theoretic conjectures in the conjecture sources (e.g., WOWII graph conjectures require symmetry groups, Noether number conjectures require finite group actions).

## Current behavior

- `finite_abelian_group.exact_factorization.compute` is the only group operation.
- `graph.symmetry.generator_orbs.compute` computes symmetry orbits but not group structure.
- There is no operation to compute the order of an element in a finitely presented group, enumerate subgroups, or compute conjugacy classes.
- The conjecture sources (Graffiti/WOWII, invariant theory) need group actions, element orders, and subgroup relationships.

## Requested capability

Add a family of bounded exact finite group operations backed by GAP or SymPy's combinatorics module:

- `group.element_order.compute`: compute the order of one element in a finitely presented or permutation group.
- `group.subgroup_lattice.compute`: enumerate subgroups of a bounded finite group.
- `group.conjugacy_classes.compute`: compute conjugacy classes and class sizes.
- `group.order.compute`: compute the order of a finitely presented or permutation group.

Each operation:
- accepts a bounded group presentation (permutation group, finitely presented group with declared generators and relators, or abelian group in Smith normal form);
- returns the exact invariant with explicit resource bounds;
- is backed by a maintained library (GAP via subprocess, or SymPy `PermutationGroup`) — not hand-rolled group theory;
- fails closed on unsupported group types, infinite groups, or budget exhaustion.

## Why this is in scope

Jacobian's AGENTS.md states that operations may use SymPy or another maintained library as a private computational engine. Computing a group invariant is an atomic operation that returns a concrete mathematical value, not a workflow or durable state. These are fundamental building blocks that compose with existing graph symmetry and finite-field operations.

## Acceptance criteria

- [ ] At least one operation (e.g., `group.element_order.compute`) is supported with a producer.
- [ ] Tests cover a cyclic group, a symmetric group, and an element of maximal order.
- [ ] The operation fails closed on unsupported group types and budget exhaustion.
- [ ] The operation is discoverable via `math.find` and executable via `math.run`.

## Related

- Existing: `finite_abelian_group.exact_factorization.compute`, `graph.symmetry.generator_orbs.compute`
- Conjecture sources: WOWII graph conjectures, Noether number conjectures, invariant theory

### PR DESCRIPTION
**Pull request by morluto**
## Problem

Jacobian had `group.order.compute`, `group.element_order.compute`, and
`group.orbit.compute` for permutation groups, but no operation for **conjugacy
classes** — a fundamental finite-group invariant needed for group-theoretic
conjectures (WOWII graph symmetry groups, Noether numbers, character theory).

Closes #1664 (conjugacy classes portion).

## Solution

Add `group.conjugacy_classes.compute`: given a permutation group by generators,
return its conjugacy classes (the partition of the group into conjugacy classes)
as permutation array forms, via SymPy's `PermutationGroup.conjugacy_classes`.
Class sizes are the orbit sizes under conjugation, composable with
`group.order.compute` (the class equation: `|G| = Σ class sizes`).

The subgroup-lattice portion of #1664 is left as a focused follow-up; this PR
delivers the conjugacy-classes invariant, which is the cleanly-exposable piece.

## Library choices

SymPy `PermutationGroup.conjugacy_classes()`, a maintained, exact kernel. No
new backend dependency; reuses the existing `PermutationGroupRequest`-style
value (generators on `{0,…,n-1}`, degree ≤ 64).

## Testing

- `make check` — Ruff, mypy, and Lean-free math/catalog/integration owners pass (3064).
- Owning tests cover: `S3` has three classes of sizes 1, 2, 3 (identity,
  3-cycles, transpositions) partitioning the 6-element group with the identity
  as its own class; and an abelian group (`C3`) where every element is its own
  class.
- Catalog conformance: the advertised example executes, re-validates, and
  survives the boundary-mutation contract test.

## Public operation admission

- Concrete gap: no conjugacy-classes operation existed.
- Why existing operations are insufficient: order/element-order/orbit do not partition the group under conjugation.
- Stable mathematical result: the conjugacy-class partition.
- Admission decision: `KEEP`.

## Public operation contract

- Mathematical input domain: a permutation group by generators on `{0,…,n-1}`, degree ≤ 64.
- Canonical public value type: reuses the generator-permutation value.
- Degenerate inputs: the trivial group has one class (the identity).
- Deterministic work bound: SymPy conjugacy classes over ≤ 64-degree groups.
- Result type: `GroupConjugacyClassesResult` (classes of array forms + method).
- Reconstruction/defining invariant: classes partition the group (`Σ sizes = |G|`, tested); abelian groups have all-singleton classes.
- Typed execution failures: non-permutation generators rejected before computation.

## Checklist

- [x] `make check` passes
- [x] Public operation changes retain owner-local admission decisions
- [x] Execution outcomes do not claim mathematical conclusions

[Continue this on Linzumi](https://serve.linzumi.com/w/williamlin1327-workspace/thread/general/0cecb73b-2b36-4041-b659-1a3441039ed1)

### CODE REVIEW

File: src/jacobian/math/group/operations.py [hunk 1]
```diff
@@ -45,3 +50,41 @@ def group_orbit(degree: int, generators: list[list[int]], point: int) -> list[in
...
+
+    order = int(group.order())
+    if order > MAX_CONJUGACY_CLASSES_GROUP_ORDER:
+        raise ValueError(
+            f"group order {order} exceeds the bounded maximum "
+            f"{MAX_CONJUGACY_CLASSES_GROUP_ORDER} for conjugacy classes "
+            f"(would materialize |G|={order} elements)"
+        )
+    classes = group.conjugacy_classes()
+    return [[list(p.array_form) for p in cls] for cls in classes]
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/group/operations.py]: Root cause: SymPy's conjugacy_classes() returns each class as a Python set, and both the native operation and the result conversion iterated those sets directly, so serialization order depended on hash randomization (reproduced with PYTHONHASHSEED=1 vs 424242 producing different tuple orders for the same S3 partition).

Fix (df4cfd5b4): group_conjugacy_classes now sorts members lexicographically within each class and sorts classes by their smallest member before returning; GroupConjugacyClassesResult additionally enforces this canonical ordering in its model validator, so every construction path serializes identically. Tool description advertises the canonical ordering (kept under the 512-char descriptor limit).

Validation: tests/math/group — new test_group_conjugacy_classes_canonical_order_s3 (exact canonical S3 partition) and test_group_conjugacy_classes_serialization_is_hash_seed_independent (subprocess repro across PYTHONHASHSEED values); plus ruff check/format, mypy, pytest tests/math/group + tests/catalog + tests/dispatch all green.

File: src/jacobian/math/group/_models.py [hunk 2]
```diff
@@ -84,3 +91,72 @@ class GroupOrbitResult(StrictModel):
...
+    classes: tuple[tuple[tuple[int, ...], ...], ...]
+    method: Literal["SYMPY_CONJUGACY_CLASSES"] = "SYMPY_CONJUGACY_CLASSES"
+
+    @model_validator(mode="after")
+    def require_nonempty_classes(self) -> Self:
+        if not self.classes:
+            raise ValueError("conjugacy classes must be nonempty")
+        for cls in self.classes:
+            if not cls:
+                raise ValueError("each conjugacy class must be nonempty")
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/group/_models.py]: Root cause: GroupConjugacyClassesResult only checked outer/inner nonemptiness, so non-permutations ((0,0),), duplicate occurrences across classes, and arbitrary non-group element sets were accepted as authoritative exact partitions — reproduced directly before the fix.

Fix (df4cfd5b4): the result model now replays the full defining invariant in require_conjugacy_class_partition: schema-visible bounds via Annotated Field constraints (permutation length ≤ degree cap, class/total sizes ≤ the 5000-order bound), then common degree, bijectivity of every element, global uniqueness, canonical ordering (strictly increasing members, classes ordered by representative), closure under composition by reconstructing the group from all listed elements and requiring |<S>| == |S| (a finite subset of a symmetric group is a subgroup iff closed), and finally each class must equal its representative's conjugation orbit in the reconstructed group (computed exactly by BFS over a minimal generator subset found via Schreier-Sims order growth — avoids SymPy's slow conjugacy_classes() near the bound, ~1.5s overhead at order 4096 vs 123s).

Validation: tests/math/group adds adversarial tests for every reviewer counterexample plus wrong-partition-of-a-valid-group, mixed degrees, unsorted members/classes, trivial/abelian boundary cases, and a JSON round-trip revalidation test; ruff check/format, mypy, pytest tests/math/group + tests/catalog + tests/dispatch all green.
